test/membrane_flac_plugin/integration_test.exs

```diff
@@ -81,4 +81,70 @@ defmodule Membrane.FLAC.Parser.IntegrationTest do
   test "fail when parsing file with junk at the end without streaming mode" do
     assert_parsing_failure("noise_and_junk", false)
   end
+
+  test "generate_best_effort_timestamps false" do
+    pipeline = prepare_pts_test_pipeline(false)
+    assert_start_of_stream(pipeline, :sink)
+
+    Enum.each(0..31, fn _index ->
+      assert_sink_buffer(pipeline, :sink, %Membrane.Buffer{pts: nil})
+    end)
+
+    assert_end_of_stream(pipeline, :sink)
+    Pipeline.terminate(pipeline)
+  end
+
+  test "generate_best_effort_timestamps true" do
+    pipeline = prepare_pts_test_pipeline(true)
+    assert_start_of_stream(pipeline, :sink)
+
+    Enum.each(0..3, fn _x ->
+      assert_sink_buffer(pipeline, :sink, _)
+    end)
+
+    Enum.each(0..27, fn index ->
+      assert_sink_buffer(pipeline, :sink, %Membrane.Buffer{pts: out_pts})
+      assert out_pts == index * 72_000_000
+    end)
+
+    assert_end_of_stream(pipeline, :sink)
+    Pipeline.terminate(pipeline)
+  end
+
+  defp prepare_pts_test_pipeline(generate_best_effort_timestamps?) do
+    import Membrane.ChildrenSpec
+
+    spec = [
+      child(:source, %Membrane.Testing.Source{output: buffers_from_file()})
+      |> child(:parser, %Membrane.FLAC.Parser{
+        generate_best_effort_timestamps?: generate_best_effort_timestamps?
+      })
+      |> child(:sink, Membrane.Testing.Sink)
+    ]
+
+    Membrane.Testing.Pipeline.start_link_supervised!(spec: spec)
+  end
+
+  defp buffers_from_file() do
+    binary = File.read!("../fixtures/noise.flac" |> Path.expand(__DIR__))
+
+    split_binary(binary)
+    |> Enum.map(fn payload ->
+      %Membrane.Buffer{
+        payload: payload,
+        pts: nil
+      }
+    end)
+  end
+
+  @spec split_binary(binary(), list(binary())) :: list(binary())
+  def split_binary(binary, acc \\ [])
+
+  def split_binary(<<binary::binary-size(2048), rest::binary>>, acc) do
+    split_binary(rest, acc ++ [binary])
+  end
+
+  def split_binary(rest, acc) when byte_size(rest) <= 2048 do
+    Enum.concat(acc, [rest])
+  end
```

Comment:
```suggestion
  def split_binary(<<binary::binary-size(2048), rest::binary>>, acc) do
    split_binary(rest, [binary] ++ acc)
  end

  def split_binary(rest, acc) when byte_size(rest) <= 2048 do
    Enum.reverse(acc) ++ [rest]
  end
```
This way you avoid doing `acc ++ [binary]` what has liner complexity, because in Elixir adding element to the end of the list requires modification of all of its elements